Code/Framework/AzCore/AzCore/DOM/DomPatch.cpp

```diff
@@ -0,0 +1,968 @@
+            return m_domPath == rhs.m_domPath && GetSourcePath() == rhs.GetSourcePath();
+        case Type::Move:
+            return m_domPath == rhs.m_domPath && GetSourcePath() == rhs.GetSourcePath();
+        case Type::Test:
+            return m_domPath == rhs.m_domPath && Utils::DeepCompareIsEqual(GetValue(), rhs.GetValue());
+        default:
+            AZ_Assert(false, "PatchOperation::GetDomRepresentation: invalid patch type specified");
+            return false;
+        }
+    }
+
+    bool PatchOperation::operator!=(const PatchOperation& rhs) const
+    {
+        return !operator==(rhs);
+    }
+
+    PatchOperation::Type PatchOperation::GetType() const
+    {
+        return m_type;
+    }
+
+    void PatchOperation::SetType(Type type)
+    {
+        m_type = type;
+    }
+
+    const Path& PatchOperation::GetDestinationPath() const
+    {
+        return m_domPath;
+    }
+
+    void PatchOperation::SetDestinationPath(Path path)
+    {
+        m_domPath = path;
+    }
+
+    const Value& PatchOperation::GetValue() const
+    {
+        return AZStd::get<Value>(m_value);
+    }
+
+    void PatchOperation::SetValue(Value value)
+    {
+        m_value = AZStd::move(value);
+    }
+
+    const Path& PatchOperation::GetSourcePath() const
+    {
+        return AZStd::get<Path>(m_value);
+    }
+
+    void PatchOperation::SetSourcePath(Path path)
+    {
+        m_value = AZStd::move(path);
+    }
+
+    AZ::Outcome<Value, AZStd::string> PatchOperation::Apply(Value rootElement) const
+    {
+        PatchOutcome outcome = ApplyInPlace(rootElement);
+        if (!outcome.IsSuccess())
+        {
+            return AZ::Failure(outcome.TakeError());
+        }
+        return AZ::Success(AZStd::move(rootElement));
+    }
+
+    PatchOperation::PatchOutcome PatchOperation::ApplyInPlace(Value& rootElement) const
+    {
+        switch (m_type)
+        {
+        case Type::Add:
+            return ApplyAdd(rootElement);
+        case Type::Remove:
+            return ApplyRemove(rootElement);
+        case Type::Replace:
+            return ApplyReplace(rootElement);
+        case Type::Copy:
+            return ApplyCopy(rootElement);
+        case Type::Move:
+            return ApplyMove(rootElement);
+        case Type::Test:
+            return ApplyTest(rootElement);
+        }
+        return AZ::Failure<AZStd::string>("Unsupported DOM patch operation specified");
+    }
+
+    Value PatchOperation::GetDomRepresentation() const
+    {
+        Value serializedPatch(Dom::Type::Object);
+        switch (m_type)
+        {
+        case Type::Add:
+            serializedPatch["op"].SetString("add");
+            serializedPatch["path"].CopyFromString(GetDestinationPath().ToString());
+            serializedPatch["value"] = GetValue();
+            break;
+        case Type::Remove:
+            serializedPatch["op"].SetString("remove");
+            serializedPatch["path"].CopyFromString(GetDestinationPath().ToString());
```

Comment:
nit: using m_domPath would make this a bit more readable on first look

Reply:
I use the getter because we have both source and destination path (potentially) and wanted to make the intent clear, but I'll mull it over if it's not reading correctly to you.